phys2bids/physio_obj.py

```diff
@@ -261,9 +264,18 @@ def check_trigger_amount(self, chtrig=1, thr=2.5, num_timepoints_expected=0, tr=
         LGR.info('Counting trigger points')
         # Use first derivative of the trigger channel to find the TRs,
         # comparing it to a given threshold.
-        trigger_deriv = np.diff(self.timeseries[chtrig])
-        timepoints = trigger_deriv > thr
-        num_timepoints_found = timepoints.sum()
+        if thr is not None:
```

Comment:
I agree with you @vinferrer, let's see what happens if we completely drop the first derivative approach!
(you can drop this "if" statement)

Reply:
My approach is actually different, so if no thr is given then we do the std thing, if it is given then then we input that thr directly to the function. Therefore we always use the trigger instead of the derivative for the triggers amont detection

Reply:
implemented